File: src/View/TextMenu.java
package View;

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

/**
 * Created by vladc on 08.11.2016.
 */
public class TextMenu {
    private Map<String, Command> commands;

    public TextMenu() {
        commands = new HashMap<>();
    }

    public void addCommand(Command c) {
        commands.put(c.getKey(), c);
    }

    private void printMenu() {
        for (Command cmd : commands.values()) {
            String line = String.format("%4s. %s", cmd.getKey(), cmd.getDescription());
            System.out.println(line);
        }
    }

    public void show() {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            printMenu();
            System.out.printf("Your option: ");
            String key = scanner.nextLine();
            Command cmd = commands.get(key);
            if (cmd == null) {
                System.out.println("Wrong command!");
                continue;
            }
            cmd.execute();
        }
    }
}

File: Toy-Language-Interpreter/View/TextMenu.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Toy_Language_Interpreter.View
{
    class TextMenu
    {
        private IDictionary<string, Command> commands;

        public TextMenu()
        {
            commands = new Dictionary<string, Command>();
        }

        public void AddCommand(Command c)
        {
            commands.Add(c.GetKey(), c);
        }

        public void PrintMenu()
        {
            foreach (Command cmd in commands.Values)
            {
                string line = string.Format("%4s. %s", cmd.GetKey(), cmd.GetDescription());
                Console.WriteLine(line);
            }
        }

        public void Show()
        {
            while (true)
            {
                PrintMenu();
                Console.Write("Your option: ");
                string key = Console.ReadLine();
                if (commands.ContainsKey(key) == false)
                {
                    Console.WriteLine("Wrong command!");
                    continue;
                }
                Command cmd = commands[key];
                cmd.Execute();
            }
        }
    }
}
